src/fread.c

```diff
@@ -1284,25 +1284,22 @@ int freadMain(freadMainArgs _args) {
   while (*nastr) {
     if (**nastr == '\0') {
       blank_is_a_NAstring = true;
-      // if blank is the only one, as is the default, clear NAstrings so that doesn't have to be checked
-      if (nastr==NAstrings && nastr+1==NULL) NAstrings=NULL;
-      nastr++;
-      continue;
+    } else {
+      const char *ch = *nastr;
+      size_t nchar = strlen(ch);
+      if (isspace(ch[0]) || isspace(ch[nchar-1]))
+        STOP(_("freadMain: NAstring <<%s>> has whitespace at the beginning or end"), ch);
+      if (strcmp(ch,"T")==0    || strcmp(ch,"F")==0 ||
+          strcmp(ch,"TRUE")==0 || strcmp(ch,"FALSE")==0 ||
+          strcmp(ch,"True")==0 || strcmp(ch,"False")==0)
+        STOP(_("freadMain: NAstring <<%s>> is recognized as type boolean, this is not permitted."), ch);
+      if ((strcmp(ch,"1")==0 || strcmp(ch,"0")==0) && args.logical01)
+        STOP(_("freadMain: NAstring <<%s>> and logical01=%s, this is not permitted."), ch, args.logical01 ? "TRUE" : "FALSE");
```

Comment:
slightly confused by the templating in the string here am I missing something? in the error case logical01 is always TRUE?

Reply:
good spot. agree logical01=TRUE could be in the string and the ternary isn't needed.